test-utils/testlib/src/standard_test_cases.rs

```diff
@@ -108,6 +108,26 @@ pub fn test_smart_contract_simple(node: impl Node) {
     validate_tx_result(node_user, root, &hash, 2);
 }
 
+pub fn test_smart_contract_self_call(node: impl Node) {
```

Comment:
Should we have one more test that uses access key to sign the transaction?

Reply:
Yes. We also need tests to verify account balances post execution depending whether it failed or succeeded.